File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cs
using System;

public class Solution 
{
    public bool solution(int x) 
    {
        string s = x.ToString();
        char[] c = s.ToCharArray();
        int[] ans = new int[c.Length];
        
        for(int i=0; i<c.Length; i++)
        {
            ans[i] = c[i]-'0';
        }
        
        int a = 0;
        foreach(int i in ans)
        {
            a += i;
        }
        return x % a == 0;
    }
}

File: 프로그래머스/1/12947. 하샤드 수/하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) 
{
    string strX = to_string(x);
    
    int s = 0;
    for(char &c : strX)
    {
        s += c - '0';
    }
    
    return (x%s == 0) ? true : false;
}